projects/lesson-2/runtime/src/substratekitties.rs

```diff
@@ -0,0 +1,113 @@
+use support::{decl_storage, decl_module, decl_event, StorageValue, StorageMap,
+              dispatch::Result, ensure};
+use system::ensure_signed;
+use sr_primitives::traits::Hash;
+use codec::{Encode, Decode};
+use runtime_io::blake2_128;
+
+#[derive(Encode, Decode, Default, Clone, PartialEq)]
+pub struct Kitty<Hash> {
+    id: Hash,
+    dna: [u8; 16],
+//    price: Balance,
+    gen: u64,
+}
+
+pub trait Trait: balances::Trait {
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+
+
+decl_storage! {
+    trait Store for Module<T: Trait> as KittyStorage {
+        Kitties get(kitty): map T::Hash => Kitty<T::Hash>;
+        KittyOwner get(owner_of): map T::Hash => Option<T::AccountId>;
+
+        AllKittiesArray get(kitty_by_index): map u64 => T::Hash;
+        AllKittiesCount get(all_kitties_count): u64;
+        AllKittiesIndex: map T::Hash => u64;
+
+        OwnedKittiesArray get(kitty_of_owner_by_index): map (T::AccountId, u64) => T::Hash;
+        OwnedKittiesCount get(owned_kitty_count): map T::AccountId => u64;
+        OwnedKittiesIndex: map T::Hash => u64;
+
+        Nonce: u128;
```

Comment:
u64 should be enough for our usage. Check the [max value](https://doc.rust-lang.org/std/primitive.u64.html#examples-1)